chrome/android/java/src/org/chromium/chrome/browser/url_button/UrlButtonController.java

```diff
@@ -0,0 +1,90 @@
+/*
+TODO:
+- fix button disappearing when user changes to another app,
+  then back to Chromium
+- use translation for "tab not selected"
+- view.animate() possibly could be used for this
+*/
+
+package org.chromium.chrome.browser.url_button;
+
+import android.content.Context;
+import android.view.MotionEvent;
+import android.view.View;
+
+import org.chromium.chrome.R;
+import org.chromium.chrome.browser.tab.Tab;
+import org.chromium.chrome.browser.tabmodel.TabModelSelector;
+import org.chromium.chrome.browser.ui.messages.snackbar.Snackbar;
+import org.chromium.chrome.browser.ui.messages.snackbar.SnackbarManager;
+import org.chromium.chrome.browser.util.ChromeAccessibilityUtil;
+import org.chromium.ui.widget.ViewRectProvider;
+
+public class UrlButtonController implements SnackbarManager.SnackbarController {
+    private final TabModelSelector mTabModelSelector;
+    private final SnackbarManager.SnackbarManageable mSnackbarManagable;
+    private final Context mContext;
+    private UrlButtonBubble mFollowBubble;
```

Comment:
It should be called `mUrlButtonBubble`, but I forgot to change it.

Also this whole private field section could be reordered a little